wled00/FX_fcn.cpp

```diff
@@ -763,47 +757,38 @@ void IRAM_ATTR_YN Segment::setPixelColor(int i, uint32_t col)
+        setPinwheelParameters(i, vW, vH, posx, posy, inc_x, inc_y, maxX, maxY);
+        unsigned totalSteps = getPinwheelSteps(vW, vH);
+        static int pixelsdrawn;
+        /*
+        Note on the skipping algorithm:
+          - in the center, the angle is way too small for efficient drawing, pixels get overdrawn a lot, making it slow
+          - tracking the radius and deciding when to actually draw pixels significantly reduces the number of overdraws
+          - the number of angular steps required for a certain radius is in theory 2*pi*radius*sqrt(2) (worst case, in the corner)
+          - we can exploit that and only draw rays that are larger than a minimum step size (need to account for rounding errors)
+        */
         // avoid re-painting the same pixel
         int lastX = INT_MIN; // impossible position
-        int lastY = INT_MIN; // impossible position
-        // draw line at angle, starting at center and ending at the segment edge
-        // we use fixed point math for better speed. Starting distance is 0.5 for better rounding
-        // int_fast16_t and int_fast32_t types changed to int, minimum bits commented
-        int posx = (centerX + 0.5f * cosVal) * Fixed_Scale; // X starting position in fixed point 18 bit
-        int posy = (centerY + 0.5f * sinVal) * Fixed_Scale; // Y starting position in fixed point 18 bit
-        int inc_x = cosVal * Fixed_Scale; // X increment per step (fixed point) 10 bit
-        int inc_y = sinVal * Fixed_Scale; // Y increment per step (fixed point) 10 bit
-
-        int32_t maxX = vW * Fixed_Scale; // X edge in fixedpoint
-        int32_t maxY = vH * Fixed_Scale; // Y edge in fixedpoint
-
-        // Odd rays start further from center if prevRay started at center.
-        static int prevRay = INT_MIN; // previous ray number
-        if ((i % 2 == 1) && (i - 1 == prevRay || i + 1 == prevRay)) {
-          int jump = min(vW/3, vH/3); // can add 2 if using medium pinwheel 
-          posx += inc_x * jump;
-          posy += inc_y * jump;
-        }
-        prevRay = i;
-
+        int lastY = INT_MIN;
+        unsigned currentR = 0; // current radius in "pixels"
         // draw ray until we hit any edge
-        while ((posx >= 0) && (posy >= 0) && (posx < maxX)  && (posy < maxY))  {
-          // scale down to integer (compiler will replace division with appropriate bitshift)
-          int x = posx / Fixed_Scale;
-          int y = posy / Fixed_Scale;
-          // set pixel
-          if (x != lastX || y != lastY) setPixelColorXY(x, y, col);  // only paint if pixel position is different
+        while ((posx < maxX)  && (posy < maxY))  {
+          int x = posx >> Fixed_Shift;
+          int y = posy >> Fixed_Shift;
+          if (x != lastX || y != lastY) { // only paint if pixel position is different
+            currentR++;
+            int requiredsteps = currentR * 8; // empirically found value (R is a rough estimation, always >= actual R)
```

Comment:
@Brandon502 can you try with `requiredsteps` set to 9 instead of 8?

Reply:
Tried 9, 10, and 12. I haven't seen any holes using 12 up to 64x64.

Reply:
the larger the number, the more overdraws and lower FPS, there are two more parameters to tweak, I tried a few options comparing holes vs FPS and current state was my best finding on 32x32 but did not check many other sizes